examples/python/alerts_stateful/alerts.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2017 The Wallaroo Authors.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+#  implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+"""
+This is an example of a stateless application that takes a transaction
+and sends an alert if its value is above or below a threshold.
+"""
+
+import wallaroo
+
+
+def application_setup(args):
+    out_host, out_port = wallaroo.tcp_parse_output_addrs(args)[0]
+
+    gen_source = wallaroo.GenSourceConfig(TransactionsGenerator())
+
+    transactions = wallaroo.source("Alerts (stateful)", gen_source)
+    pipeline = (transactions
+        .key_by(extract_user)
+        .to(check_transaction_total)
+        .to_sink(wallaroo.TCPSinkConfig(out_host, out_port, encode_alert)))
+
+    return wallaroo.build_application("Alerts (stateful)", pipeline)
+
+class Transaction(object):
+    def __init__(self, user, amount):
+        self.user = user
+        self.amount = amount
+
+class TransactionTotal(object):
```

Comment:
This should be:

```python
class TransactionTotal(object):
    def __init__(self):
        self.total = 0
```

Reply:
Or am I missing something? Maybe I'm missing something.

Reply:
I only chose the other way because it was more compact (and because the constructor will always have only `self` as a parameter).  If you think that will strike Python programmers as weird, I can change it.

Reply:
See above comment on class variables, then weep because Python.

Reply:
Ouch.  Ok I'll update these and the actual examples.